common/main/java/com/couchbase/lite/internal/logging/LogSinksImpl.java

```diff
@@ -0,0 +1,298 @@
+        catch (Exception e) {
+            log(consoleLogger, LogLevel.WARNING, LogDomain.DATABASE, "File logger failure: " + Log.formatStackTrace(e));
+        }
+
+        writeToLocalLoggers(level, domain, msg);
+    }
+
+    @SuppressWarnings({"RegexpSinglelineJava", "PMD.SystemPrintln", "PMD.GuardLogStatement"})
+    public void writeToLocalLoggers(@NonNull LogLevel level, @NonNull LogDomain domain, @Nullable String message) {
+        final String msg = (message == null) ? "" : message;
+
+        final ConsoleLogSink console = consoleLogger;
+        try { log(console, level, domain, msg); }
+        catch (Exception e) { System.err.println("Console logger failure" + Log.formatStackTrace(e)); }
+
+        // A custom logger is client code: give it 1 second on a safe thread
+        final BaseLogSink custom = customLogger;
+        if ((custom != null) && ((custom.getLevel().compareTo(level) <= 0))) {
+            if (customLogQueue.getPending() > LOG_QUEUE_MAX) {
+                log(console, LogLevel.WARNING, LogDomain.DATABASE, "Log queue overflow: Logs dropped");
+                return;
+            }
+
+            customLogQueue.execute(() -> {
+                try { custom.log(level, domain, msg); }
+                catch (Exception e) {
+                    log(console, LogLevel.WARNING, LogDomain.DATABASE, "Custom log failure" + Log.formatStackTrace(e));
+                }
+            });
+        }
+    }
+
+    public boolean shouldLog(@NonNull LogLevel level, @NonNull LogDomain domain) {
+        return (logLevel.get().compareTo(level) <= 0) && logDomains.get().contains(domain);
+    }
+
+    @VisibleForTesting
+    @NonNull
+    public C4Log getC4Log() { return c4Log; }
+
+    @VisibleForTesting
+    public void setC4Log(@NonNull C4Log c4Log) { this.c4Log = c4Log; }
+
+    private void forbidNewAndLegacyLogging(@Nullable AbstractLogSink newLogger) {
+        if (newLogger == null) { return; }
+
+        final Boolean usedLegacyLogging = this.usedLegacyLogging;
+        this.usedLegacyLogging = newLogger.isLegacy();
+
+        if (usedLegacyLogging == null) { return; }
+
+        if (!usedLegacyLogging.equals(this.usedLegacyLogging)) {
+            throw new CouchbaseLiteError("Cannot mix new and legacy loggers");
+        }
+    }
+
+    private void setLogFilter() {
+        final Set<LogDomain> newDomains = new HashSet<>();
+        LogLevel callbackLevel = LogLevel.NONE;
+
+        LogLevel l;
+
+        if (consoleLogger != null) {
+            l = consoleLogger.getLevel();
+            if (l.compareTo(callbackLevel) < 0) { callbackLevel = l; }
+            newDomains.addAll(consoleLogger.getDomains());
+        }
+
+        if (customLogger != null) {
+            l = customLogger.getLevel();
+            if (l.compareTo(callbackLevel) < 0) { callbackLevel = l; }
+            newDomains.addAll(customLogger.getDomains());
+        }
+
+        // ignore the file logger's domains
```

Comment:
This would mean that newDomains will always contain all domains because file logging is for all domains?

Reply:
fixed

Reply:
You are absolutely right.  I completely misunderstood the issue.  Fixed.